staging/src/k8s.io/apiextensions-apiserver/pkg/apihelpers/helpers.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apihelpers
+
+import (
+	"net/url"
+	"strings"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+// IsGroupProtected returns whether or not a group specified for a CRD is protected for the community and needs
+// to have the v1beta1.KubeAPIApprovalAnnotation set.
+func IsGroupProtected(group string) bool {
+	switch {
+	case group == "k8s.io" || strings.HasSuffix(group, ".k8s.io"):
+		return true
+	case group == "kubernetes.io" || strings.HasSuffix(group, ".kubernetes.io"):
+		return true
+
+	default:
+		return false
+	}
+
+}
+
+// APIApprovalState covers the various options for API approval annotation states
+type APIApprovalState int
+
+const (
+	// APIApproved if the annotation has a URL (this means the API is approved)
+	APIApproved APIApprovalState = iota
+	// APIUnapproved if the annotation starts with "unapproved" indicating that for whatever reason the API isn't approved, but we should allow its creation
+	APIUnapproved
+	// APIApprovalMissing means the annotation is empty
+	APIApprovalMissing
+	// APIApprovalInvalid means the annotation doesn't have an expected value
+	APIApprovalInvalid
+)
+
+// GetAPIApprovalState returns the state of the API approval.
+func GetAPIApprovalState(annotations map[string]string) APIApprovalState {
+	annotation := annotations[v1beta1.KubeAPIApprovalAnnotation]
+
+	url, annotationURLParseErr := url.ParseRequestURI(annotation)
+	switch {
+	case len(annotation) == 0:
```

Comment:
I guess you check that the call above works. Would though put that above the parse call for clarity.